packages/core/src/storages/request_list_adapter.ts

```diff
@@ -0,0 +1,102 @@
+import type { Dictionary } from '@crawlee/types';
+
+import type { Request } from '../request';
+import type { IRequestList } from './request_list';
+import type {
+    AddRequestsBatchedResult,
+    IRequestManager,
+    RequestQueueOperationInfo,
+    RequestQueueOperationOptions,
+} from './request_provider';
+
+/**
+ * Adapts the IRequestList interface to the IRequestManager interface.
+ * It simply throws an exception when inserting requests is attempted.
+ * @internal
+ */
+export class RequestListAdapter implements IRequestManager {
+    constructor(private requestList: IRequestList) {}
+
+    /**
+     * @inheritdoc
+     */
+    async isFinished(): Promise<boolean> {
+        return this.requestList.isFinished();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async isEmpty(): Promise<boolean> {
+        return this.requestList.isEmpty();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async handledCount(): Promise<number> {
+        return Promise.resolve(this.requestList.handledCount());
+    }
+
+    /**
+     * @inheritdoc
+     */
+    getTotalCount(): number {
+        return this.requestList.length();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    getPendingCount(): number {
+        return this.requestList.length() - this.requestList.handledCount();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async fetchNextRequest<T extends Dictionary = Dictionary>(): Promise<Request<T> | null> {
+        return await this.requestList.fetchNextRequest();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async markRequestHandled(request: Request): Promise<void> {
+        return this.requestList.markRequestHandled(request);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    async reclaimRequest(
+        request: Request,
+        _options?: RequestQueueOperationOptions,
```

Comment:
The `options` argument gets dropped silently here. Maybe log a warning `if(options)`? Or at least add a JSDoc comment about this?

Reply:
I wouldn't fret about this too much, it's an internal class that will be removed in v4.